File: example/src/main/java/com/github/vivchar/example/pages/simple/items/ViewStateViewModel.java
package com.github.vivchar.example.pages.simple.items;

import android.support.annotation.NonNull;

import com.github.vivchar.rendererrecyclerviewadapter.DefaultCompositeViewModel;
import com.github.vivchar.rendererrecyclerviewadapter.ViewModel;

import java.util.List;

/**
 * Created by Vivchar Vitaly on 28.12.17.
 */

public class ViewStateViewModel extends DefaultCompositeViewModel {

	private final int mID;

	public ViewStateViewModel(final int ID, @NonNull final List<? extends ViewModel> items) {
		super(items);
		mID = ID;
	}

	public int getID() {
		return mID;
	}
}

File: example/src/main/java/com/github/vivchar/example/pages/simple/ViewStateViewModel.kt
package com.github.vivchar.example.pages.simple

import androidx.lifecycle.ViewModel
import com.github.vivchar.rendererrecyclerviewadapter.ViewModel as ItemModel
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow

data class ViewStateUiState(val items: List<ItemModel>)

class ViewStateViewModel : ViewModel() {

	private val dataProvider = YourDataProvider()
	private val _state = MutableStateFlow(ViewStateUiState(items = dataProvider.stateItems))
	val state: StateFlow<ViewStateUiState> = _state
}
